rotkehlchen/chain/ethereum/modules/polygon_pos_bridge/decoder.py

```diff
@@ -0,0 +1,155 @@
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label is not None and
+                self.base.is_tracked(string_to_evm_address(event.location_label)) and
+                event.address in {BRIDGE_ADDRESS, ERC20_BRIDGE_ADDRESS, PLASMA_BRIDGE_ADDRESS}
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.notes = f'Bridge {event.balance.amount} {event.asset.resolve_to_asset_with_symbol().symbol} from Ethereum to Polygon PoS'  # noqa: E501
+                event.counterparty = CPT_POLYGON
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_withdraw(self, context: DecoderContext) -> DecodingOutput:
+        """Decode ETH and ERC20 bridge deposit events. (not Plasma, see _decode_plasma_withdraw)
+
+        Mapped to BRIDGE_ADDRESS, ETH_BRIDGE_ADDRESS, and ERC20_BRIDGE_ADDRESS. Also called
+        from _decode_plasma_process_exit. Checks if the topic indicates a withdrawal has been
+        initiated, and then modifies the corresponding receive event.
+        """
+        if context.tx_log.topics[0] not in {EXITED_ERC20_TOPIC, EXITED_ETHER_TOPIC, PROCESS_EXIT_TOPIC}:  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label is not None and
+                self.base.is_tracked(string_to_evm_address(event.location_label)) and
+                event.address in {BRIDGE_ADDRESS, ETH_BRIDGE_ADDRESS, ERC20_BRIDGE_ADDRESS, PLASMA_BRIDGE_ADDRESS}  # noqa: E501
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.notes = f'Bridge {event.balance.amount} {event.asset.resolve_to_asset_with_symbol().symbol} from Polygon PoS to Ethereum'  # noqa: E501
+                event.counterparty = CPT_POLYGON
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _is_exit_nft_event(self, event: 'EvmEvent', match_event_type: HistoryEventType) -> bool:
+        """Check if the event matches the specified type and involves the Exit NFT."""
+        return (
+            event.event_type == match_event_type and
+            event.event_subtype == HistoryEventSubType.NONE and
+            event.address == ZERO_ADDRESS and
+            event.asset.resolve_to_evm_token().token_kind == EvmTokenKind.ERC721
+        )
+
+    def _decode_plasma_start_exit(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes Plasma bridge start exit events."""
+        for event in context.decoded_events:
```

Comment:
This function is only for NFTs right? You should state it in the name and docstring